lib/smack/SplitStructLoadStore.cpp

```diff
@@ -0,0 +1,148 @@
+#include "smack/SplitStructLoadStore.h"
+#include "llvm/IR/InstIterator.h"
+
+namespace smack {
+
+using namespace llvm;
+
+std::vector<Value*> getFirsts(std::vector<std::pair<Value*, unsigned> > lst)
+{
+    std::vector<Value*> ret;
+    for (auto p = lst.begin(); p != lst.end(); ++p)
+      ret.push_back(std::get<0>(*p));
+    return ret;
+}
+
+std::vector<unsigned> getSeconds(std::vector<std::pair<Value*, unsigned> > lst)
+{
+    std::vector<unsigned> ret;
+    for (auto p = lst.begin()+1; p != lst.end(); ++p)
```

Comment:
Wait, what happens here if `lst.begin == lst.end`?? Meaning, when you have only one element in this list. Is that situation possible?
If not, then it would be good to add an assertion here to make sure.